### ISSUE TITLE
[Web] Ruffle does not work with SWFObject

### ISSUE THREAD

**Issue by Toad06** [ruffle-rs/ruffle#448]
Ruffle will fail to run any Flash file using SWFObject. For example, the following code will not work (unless Flash is enabled... but that's not really useful 😜):

> swfobject.embedSWF("uniwar.swf", "file_div", "550", "400", "8");

In the attached sample:
- Add a nightly build in "ruffle/".
- Make sure Flash is disabled and run "test.html". You should see "Flash Player not detected!" instead of the "Click to play" by Ruffle.
- Eventually run "extension.html" using the browser's extension, you will notice the same issue.
- Run "workaround.html", things should work as expected because swfobject.embedSWF() has been rewritten and using "const swfobject" prevents SWFObject to run (but that will trigger an error in the console... 😅).

[swfobjectTest.zip](https://github.com/ruffle-rs/ruffle/files/4338003/swfobjectTest.zip)

**Comment by Herschel** [comment]
https://github.com/swfobject/swfobject/blob/master/swfobject/src/swfobject.js#L41 is the SWFObject code that detects the Flash Player version. This runs before the Ruffle code that installs the fake plugin, because Ruffle will not install polyfills until the `DOMContentLoaded` event. We should at least install the plugin spoof immediately when Ruffle is initialized.

Secondly, Ruffle returns "Shockwave Flash 32.0 r0 (compatible; Ruffle 0.0.1)" for the plugin version, but this breaks SWFObject's version parsing at https://github.com/swfobject/swfobject/blob/master/swfobject/src/swfobject.js#L58-L60. We should just return the standard version string; people could detect a Ruffle install in other ways if they needed to.

**Comment by ROBERT-MCDOWELL** [comment]
@Herschel 
whiy not to make a PR to make swfObject compatible to ruffle?

**Comment by Herschel** [comment]
This should work now as long as the Ruffle script tag is above the SWFObject script tag (we should document this).

SWFObject does spit out a JS error in the console, but it seems okay -- it tries to call object.GetVariable on a fake flash embed for a more sophisticated version check, but Ruffle would have already removed the object.

**Comment by Toad06** [comment]
Tested with today's JS build (2020-03-27), it works as described above in Firefox and Chrome.

However I've not been able to make it using the extension. I'm not sure if it's as intended, or if it requires some extra configuration.

**Comment by Toad06** [comment]
@Herschel ,
This is still not working using the extension. I've found out that if you add this in "manifest.json":
`"run_at": "document_start"`
then it will work most of the time in Firefox. Unfortunately, it's the opposite in Chrome, as this will only (and randomly) work when caching is disabled.

I've tried some things to workaround this... with no success at all. What I've tried:
- editing lv0.js and the way it appends ruffle.js to the DOM, so it no longer uses "async" and "await".
```
let scriptelem = document.createElement("script");
scriptelem.setAttribute("src", ext_path + "dist/ruffle.js");
scriptelem.setAttribute("data-Ruffle_runtime_path", ext_path); // ruffle.js has also been edited to get this value
scriptelem.setAttribute("data-Ruffle_obfuscated_event_prefix", obfuscated_event_prefix); // this one as well
document.head.appendChild(scriptelem);
```
This is working pretty fine with Firefox but Chrome says "TypeError: Cannot read property 'appendChild' of null". Replacing _document.head_ with _document.documentElement_ fixes the crash but that's all it is.

- using a Mutation Observer and appending the Ruffle script as soon as the head node is added. Again, this has not changed anything.

I'm giving up for now. 😅
